js/pages/vocabulary/components/search.js

```diff
@@ -540,7 +537,10 @@ define([
 			const concepts = commonUtils.getSelectedConcepts(this.data);
 			const items = commonUtils.buildConceptSetItems(concepts, options);
 			const itemsToAdd = items.map(item => new ConceptSetItem(item));
-			this.previewConcepts(itemsToAdd.concat(this.conceptSetStore.current() ? this.conceptSetStore.current().expression.items() : []));
+			const existingConceptsCopy = this.conceptSetStore.current()
+				? this.conceptSetStore.current().expression.items().map(item => new ConceptSetItem(item))
+				: [];
+			this.previewConcepts(itemsToAdd.concat(existingConceptsCopy));
```

Comment:
Here, you would just say:

```
const existingConceptsCopy = new ConceptSet(ko.toJS(this.conceptSetStore.current()));
```

Although what i'm suggesting above assumed that the ConceptSetPreview would be working off of a ConceptSet and not an array of ConceptSetItems...which is fine but in that case, I should say that the change is:

```
const existingConceptsCopy = this.conceptSetStore.current()
				? this.conceptSetStore.current().expression.items().map(item => new ConceptSetItem(ko.toJS(item)))
				: [];
```

And then, when the 'Apply' button is clicked in the preview editor, that will overwrite this.conceptSetStore.current() with the new state of the `ConceptSetItem`s.

But, overall, you have the right approach:   you need to clone the existing concept set for the user to modify, and only 'commit' the change when they click 'Apply' and discard those changes (ie: do nothing) if they cancel.

Reply:
Note, that last part (where when the click is made, the finalized change is saved back to the current concept set...if we're working off a clone, doesn't the changes made to the clone need to propagate tack to the object?

Reply:
Fixed. 
Cloned concepts are concatenated with selected in the search results. They all stored in this.previewConcepts() and passed to current concept set in the addPreviewConcepts() method.